src/array/map-reduce.jl

```diff
@@ -40,29 +40,29 @@ end
 
 function stage(ctx::Context, r::ReduceBlock)
     inp = stage(ctx, r.input)
-    reduced_parts = map(x -> Thunk(r.op, nothing=>x; get_result=r.get_result), chunks(inp))
-    Thunk((xs...) -> r.op_master(xs), map(part->nothing=>part, reduced_parts)...; meta=true)
+    reduced_parts = map(x -> (Dagger.@spawn get_result=r.get_result r.op(x)), chunks(inp))
+    r_op_master(args...,) = r.op_master(args)
+    Dagger.@spawn meta=true r_op_master(reduced_parts...) 
 end
 
 reduceblock_async(f, x::ArrayOp; get_result=true) = ReduceBlock(f, f, x, get_result)
 reduceblock_async(f, g::Function, x::ArrayOp; get_result=true) = ReduceBlock(f, g, x, get_result)
 
-reduceblock(f, x::ArrayOp) = compute(reduceblock_async(f, x))
-reduceblock(f, g::Function, x::ArrayOp) =
-    compute(reduceblock_async(f, g, x))
+reduceblock(f, x::ArrayOp) = reduceblock_async(f, x)
+reduceblock(f, g::Function, x::ArrayOp) = reduceblock_async(f, g, x)
```

Comment:
Maybe let's move the `fetch` calls below back up to these two definitions (since that will cover all instances of `reduce`)